src/instances.ts

```diff
@@ -317,6 +321,21 @@ function successfulTypeScriptInstance(
   return { instance };
 }
 
+/** Update the set of root filenames in the instance from a parsed tsconfig.json */
+export function updateInstanceRootFileNames(
+  loaderOptions: LoaderOptions,
+  instance: TSInstance,
+  configParseResult: typescript.ParsedCommandLine
+) {
+  const files = loaderOptions.onlyCompileBundledFiles
+    ? configParseResult.fileNames.filter(fileName =>
```

Comment:
I thiiiink this is wrong? `onlyCompileBundledFiles` != _only_ load declaration files, and you never add in the ones Webpack is specifically asking for. With the old approach, the files would eventually get added to `files` as the compilation progressed, meaning that the TypeScript program would eventually see them as root files. But now, they never get added, which means TypeScript isn’t going to know about _any_ `.ts` files. We went from adding too many files to too few (with `onlyCompileBundledFiles`).

@johnnyreilly we might be able to hack around this, but given what we've observed about needing an accurately represented set of root files, I’m highly suspicious of this loader option’s existence in the first place. I think it’s a little dangerous because it sounds like a promising thing to enable when build times are slow. “Oh, wait, is ts-loader compiling files that aren’t even in my bundle? That doesn’t sound useful, let me turn that off!” But it has a foreboding aura of “this is going to be really hard to reason about as we introduce features like:”

- project references
- incremental builds
- Yarn PNP

Reply:
Yeah - you make a good point. This is the initial PR that added it:

https://github.com/TypeStrong/ts-loader/pull/671

Do have a read and let me know if you're convinced by the need of this going forward. Perhaps it's worth removing this option if it causes more harm than good? 

I think it's s relatively niche option so it may not be a problem. Major version bump but that's fine